### PR TITLE
Bump all Knative go-deps to version 1.10

### PR DESCRIPTION
**Pull request by ReToCode**
## Proposed Changes
- Bump all Knative go-deps to version 1.10
- Update `hack/patches/005-conversion-webhook.patch` to reflect changes in 1.10
- Drop `hack/patches/013-rekt-test-add-service-port-name.patch` because it is already in 1.10
- Update `hack/patches/remove_eventing_ns_manifest.patch` to reflect changes in 1.10
- Use subdir for ingress manifests based on https://github.com/knative/operator/pull/1355

### CODE REVIEW

**Comment by ReToCode** [comment]
/cc @matzew @skonto @pierDipi 

@pierDipi @matzew  are the to patches for Eventing ok like this? Outcome: https://github.com/openshift-knative/serverless-operator/pull/2207/files#diff-426607a57d59b8ed0160ad85ad91a1edf983acb570ce2f4f57a624c07e8c41f5R27 and https://github.com/openshift-knative/serverless-operator/pull/2207/files#diff-b970ddcc07cde675e0aa5e5d6e107c3f29882ab7a8f6ecb840debebacc512b45R25

@skonto @nak3 are webhook patches ok like this? Outcome: https://github.com/openshift-knative/serverless-operator/pull/2207/files#diff-4ade17300ba49da3fe4bcf7ff0e67471ae89f4938d3d926895c0efaa7943c41bR44

**Comment by ReToCode** [comment]
@pierDipi Hm, [this](https://github.com/openshift-knative/serverless-operator/blob/main/vendor/knative.dev/eventing/test/rekt/resources/addressable/addressable.go#L36) does not work anymore, as Addressable with CACerts is no longer assignable to URL.

**Comment by nak3** [comment]
> @skonto @nak3 are webhook patches ok like this? Outcome: https://github.com/openshift-knative/serverless-operator/pull/2207/files#diff-4ade17300ba49da3fe4bcf7ff0e67471ae89f4938d3d926895c0efaa7943c41bR44

Okay to me but I guess Stavros is working on https://github.com/openshift-knative/serverless-operator/pull/2148 so defer to him.

File: go.mod
Comment (pierDipi) [inline go.mod:179]: This is pointing to main, commit `https://github.com/knative-extensions/reconciler-test/commit/a8041ba726ed`

Comment (ReToCode) [reply go.mod]: 👍 
I hope [this](https://github.com/openshift-knative/serverless-operator/pull/2207/commits/043ca97c29b7dc9acea6a952446dfdeab35be5f5#diff-5b45c8cc2cb795405146417d3efa55bc2472f989e32bef288f7493e190f1388bL101) is compatible?

Comment (ReToCode) [reply go.mod]: https://github.com/knative-extensions/reconciler-test/pull/566

**Comment by ReToCode** [comment]
>  error=Get "https://10.0.128.94:10250/containerLogs/ci-op-p6d50wjt/upstream-e2e-aws-ocp-413-serverless-e2e/test": x509: certificate signed by unknown authority

/retest

File: vendor/knative.dev/eventing/test/rekt/resources/containersource/containersource.yaml
Comment (matzew) [inline vendor/knative.dev/eventing/test/rekt/resources/containersource/containersource.yaml:52]: that's a wrong reference here, we don't do the `ko://`  this should be proper in midstream....

**Comment by matzew** [comment]
```
  - Match errors -
        0 - Kn-Namespace header not found
        1 - Kn-Namespace header not found
        2 - Kn-Namespace header not found
        3 - Kn-Namespace header not found
```

**Comment by mgencur** [comment]
> @mgencur is that above Kn-Namespace related to this?
https://github.com/openshift-knative/serverless-operator/pull/2195

I don't think it's related. The test is obviously updated in this PR so it expects the KnHeader. All the other tests are passing (channel-related ones etc. which expect the KnHeader as well). So, the Knative Forwarder component should be forwarding the header correctly. I also checked the image of Ping source adapter and it's version 1.10 which is OK.
I don't know why this error happens.

**Comment by mgencur** [comment]
@matzew I'd probably try to remove [this line](https://github.com/openshift-knative/serverless-operator/blob/main/test/eventinge2erekt/main_test.go#L36) to see if the problem is in the forwarder. If the test passes the problem is in forwarder, if it still fails it must be in PingSource.

**Comment by creydr** [comment]
In the PingSource we add since 1.9 the Kn-Namespace header (https://github.com/openshift-knative/eventing/pull/265). Not sure, if this header gets forwarded in the forwarder too

**Comment by matzew** [comment]
```
+++ b/go.mod
@@ -184,8 +184,8 @@ replace (
 	knative.dev/eventing-kafka-broker => github.com/openshift-knative/eventing-kafka-broker v0.25.1-0.20230803130431-bf8f4e608e85
 	knative.dev/hack => knative.dev/hack v0.0.0-20230417170854-f591fea109b3
 	knative.dev/networking => knative.dev/networking v0.0.0-20230419144338-e5d04e805e50
- 	knative.dev/reconciler-test => knative.dev/reconciler-test v0.0.0-202[30](https://github.com/openshift-knative/serverless-operator/actions/runs/5806025450/job/15738118955?pr=2207#step:10:31)808072333-158379f56e0b
 	knative.dev/pkg => knative.dev/pkg v0.0.0-20230418073056-dfad48eaa5d0
+	knative.dev/reconciler-test => knative.dev/reconciler-test v0.0.0-20230808072333-158379f56e0b
 	knative.dev/serving => github.com/openshift-knative/serving v0.10.1-0.20230721070749-1af735104[33](https://github.com/openshift-knative/serverless-operator/actions/runs/5806025450/job/15738118955?pr=2207#step:10:34)0
 )
```

**Comment by mgencur** [comment]
Yeah. Looking at the logs from the [forwarder](https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/origin-ci-test/pr-logs/pull/openshift-knative_serverless-operator/2207/pull-ci-openshift-knative-serverless-operator-main-4.13-operator-e2e-aws-ocp-413/1688863651936604160/artifacts/operator-e2e-aws-ocp-413/gather-extra/artifacts/pods/test-znrhfwtv_sink-bvacvnzb-00001-deployment-664fdc5f95-rwpwf_eventshub-forwarder.log):

Received event:
```
{"level":"info","ts":"2023-08-08T11:29:00.203Z","logger":"eventshub.event logger","caller":"logger_vent/logger.go:26","msg":"Event: \n-- EventInfo --\n--- Kind: Received ---\n--- Event ---\nContext Attributes,\n  specversion: 1.0\n  type: dev.knative.sources.ping\n  source: /apis/v1/namespaces/test-znrhfwtv/pingsources/pingsource-hdicxhyy\n  id: 7edb6ea3-31ea-4dbb-a5d6-62aaa9ea5bff\n  time: 2023-08-08T11:29:00.081538462Z\n\n--- HTTP headers ---\n  X-Request-Id: 56a28edf-d5bb-4e49-bd1c-9a0dd08b9b1b\n  X-B3-Spanid: d0d1dcc37b90469a\n  X-B3-Traceid: d626695bb5c8ad09af015327faa16bb3\n  K-Proxy-Request: activator\n  X-B3-Sampled: 1\n  User-Agent: Go-http-client/1.1\n  Host: sink-bvacvnzb.test-znrhfwtv.svc.cluster.local\n  Content-Length: 0\n  Traceparent: 00-d626695bb5c8ad09af015327faa16bb3-d0d1dcc37b90469a-01\n  Accept-Encoding: gzip\n  Kn-Namespace: test-znrhfwtv\n  X-Forwarded-Proto: http\n  Forwarded: for=10.131.0.23;proto=http\n  X-Forwarded-For: 10.131.0.23, 10.129.2.23\n\n--- Origin: '127.0.0.1:44260' ---\n--- Observer: 'sink-bvacvnzb' ---\n--- Time: 2023-08-08 11:29:00.203898196 +0000 UTC m=+106.857147156 ---\n--- Sequence: 0 ---\n--- Sent Id:  ' ---\n--------------------\n"}
```

Sent/forwarded event:
```
{"level":"info","ts":"2023-08-08T11:29:00.212Z","logger":"eventshub.event logger","caller":"logger_vent/logger.go:26","msg":"Event: \n-- EventInfo --\n--- Kind: Sent ---\n--- Event ---\nContext Attributes,\n  specversion: 1.0\n  type: dev.knative.sources.ping\n  source: /apis/v1/namespaces/test-znrhfwtv/pingsources/pingsource-hdicxhyy\n  id: 7edb6ea3-31ea-4dbb-a5d6-62aaa9ea5bff\n  time: 2023-08-08T11:29:00.081538462Z\n\n--- HTTP headers ---\n  Ce-Id: 7edb6ea3-31ea-4dbb-a5d6-62aaa9ea5bff\n  Ce-Source: /apis/v1/namespaces/test-znrhfwtv/pingsources/pingsource-hdicxhyy\n  Ce-Specversion: 1.0\n  Ce-Type: dev.knative.sources.ping\n  Ce-Time: 2023-08-08T11:29:00.081538462Z\n\n--- Origin: 'sink-bvacvnzb' ---\n--- Observer: 'sink-bvacvnzb' ---\n--- Time: 2023-08-08 11:29:00.212682216 +0000 UTC m=+106.865931174 ---\n--- Sequence: 0 ---\n--- Sent Id:  '7edb6ea3-31ea-4dbb-a5d6-62aaa9ea5bff ---\n--------------------\n"}
```

**Comment by mgencur** [comment]
I will work on the fix for the forwarder.

**Comment by mgencur** [comment]
Would this be acceptable? https://github.com/knative-extensions/reconciler-test/pull/572
I still need to test it with real deployment but I think it should work.

**Comment by ReToCode** [comment]
> Would this be acceptable? https://github.com/knative-extensions/reconciler-test/pull/572

Deferring this to Eventing folks. Just out of curiosity, is there a specific reason the forwarder filters headers in the first place?

**Comment by mgencur** [comment]
> Deferring this to Eventing folks. Just out of curiosity, is there a specific reason the forwarder filters headers in the first place?

No specific reason. Forwarding of headers was just not implemented, probably by mistake.

**Comment by matzew** [comment]
> No specific reason. Forwarding of headers was just not implemented, probably by mistake.

should it just be transparent? forward, as is - and not edit/manipulate the actual req?

**Comment by mgencur** [comment]
Apparently, something like this works as well: https://github.com/knative-extensions/reconciler-test/pull/575

File: test/images-rekt.yaml
Comment (mgencur) [inline test/images-rekt.yaml:1]: Is it right that you still use knative-nightly here? Should that be v1.10?
I think you said nightly cannot be used anymore. Just curious.

Comment (ReToCode) [reply test/images-rekt.yaml:1]: Not sure, I think my comments were only related to the code deps from main branch because of a conflict with the changes to Adressable upstream. We should be fine with that image (I think..., let's see).

Comment (mgencur) [reply test/images-rekt.yaml:1]: OK. Anyway, the backports helped for multiarch builds here: https://github.com/openshift-knative/serverless-operator/commit/3863526f2d8f5271f275806e100208cde5908101
We should probably use this image as well here so that we have the multiarch image from the beginning and no need to patch later like in the commit above.
Also, the version will change with each release so we probably need to create a template for this file and generate the resulting file automatically. But this can be done in a follow-up. I'd suggest using the quay image in this file for both eventshub and heartbeats. The latest available heartbeats image at quay is v1.9 though: quay.io/openshift-knative/eventing/heartbeats:v1.9 That's because the image was moved from test images to core images in 1.10 and the github action is not building it (this needs a fix!)

Comment (mgencur) [reply test/images-rekt.yaml:1]: Created https://issues.redhat.com/browse/SRVCOM-2649

**Comment by matzew** [comment]
It passed. testing it _again_

/test 4.13-operator-e2e-aws-ocp-413

**Review by matzew — APPROVED** [review]
/lgtm
/approve

/hold until the mentioned e2e test passes

**Comment by ReToCode** [comment]
Yay 🎉 

/unhold

**Pull request merged**
